### PR TITLE
fix: [PL-42355]: Modified the count metrics names in delegate metrics

### PR DESCRIPTION
**Pull request by VikasMaddukuriHarness**
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the CODEOWNERS. 
With the changes as per this jira - https://harness.atlassian.net/browse/PL-42354, we need to update some metrics names in documentation. This pr is to update those names
## What Type of PR is This?

- [x] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

**Review by brian-f-harness — APPROVED** [review]

File: docs/platform/delegates/manage-delegates/delegate-metrics.md [hunk 1]
```diff
@@ -14,10 +14,10 @@ The delegate is instrumented for the collection of the following delegate agent
 | :-- | :-- |
 | `io_harness_custom_metric_task_execution_time` | The time it takes to complete a task (in seconds). |
 | `io_harness_custom_metric_tasks_currently_executing` | The number of tasks underway. |
-| `io_harness_custom_metric_task_timeout` | The number of tasks that time out before completion. |
-| `io_harness_custom_metric_task_completed` | The number of tasks completed. |
-| `io_harness_custom_metric_task_failed` | The number of failed tasks. |
-| `io_harness_custom_metric_task_rejected`* | The number of tasks rejected because of a high load on the delegate. |
+| `io_harness_custom_metric_task_timeout_total` | The number of tasks that time out before completion. |
```
Comment (pratmit) [inline docs/platform/delegates/manage-delegates/delegate-metrics.md:17]: What about the metrics name for the users who are on an older version of delegate?

**Review by pratmit — COMMENTED** [review]
Added a comment for the older version delegate?

File: docs/platform/delegates/manage-delegates/delegate-metrics.md (hunk 1 above)
Comment (brian-f-harness) [reply docs/platform/delegates/manage-delegates/delegate-metrics.md:17]: @pratmit Good question. @VikasMaddukuriHarness Do we need to update the content to call out the delegate version dependency and have both names documented?

**Comment by brian-f-harness** [comment]
> Added a comment for the older version delegate?

Put original topic content back in place and opened a new PR: https://github.com/harness/developer-hub/pull/4164

**Pull request merged**